docs/core/integration-quality-scale/rules/iqs005.md

```diff
@@ -0,0 +1,41 @@
+---
+title: "IQS005: Support config entry unloading"
+---
+
+IQS005 is as follows:
+
+Support config entry unloading
+
+## Reasoning
+
+Sometimes a config entry ends up in a state where it is stuck.
+This can have many reasons, but sometimes the integration works again after the integration has been reloaded.
+
+In order to support this, the integration should support unloading the config entry.
+This gives users the ability to reload the integration without having to restart Home Assistant.
+Reloading an integration can be a powerful tool for users to fix issues with integrations.
+
+## Example implementation
+
+The code required to reload an integration is very little, with a lot of benefits for the user.
+
+The method that has to be added to `__init__.py` looks very similar to the `async_setup_entry` method.
+
+`__init__.py`:
+```python
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
```

Comment:
I think we need to extend this example or add another example, to show the capabilities here.

While it is true that it will look like this for most integrations, the goal here is to allow for integration code on this place as well. E.g., closing connections, unsubscribing webhooks and things like that.